scripts/o3de/o3de/repo_properties.py

```diff
@@ -0,0 +1,399 @@
+                                   repo_json['repo_uri'], 
+                                   download_prefix)
+                index = _find_index(repo_objects_data, f'{object_typename}_name', json_data[f'{object_typename}_name'])
+                # if the added gem is same with partial changes
+                if index != -1:
+                    # we want changes only
+                    version_data = _changed(repo_objects_data[index], json_data)
+                    if not version_data:
+                        # there is no difference between the gem being added and 
+                        # what already exists in gems_data
+                        continue
+
+                    # versions data must always include the version
+                    version_data['version'] = json_data['version']
+
+                    # versions data must always include one of the download/source uris
+                    if 'download_source_uri' in json_data:
+                        version_data['download_source_uri'] = json_data['download_source_uri']
+                    if 'source_control_uri' in json_data:
+                        version_data['source_control_uri'] = json_data['source_control_uri']
+                    if 'source_control_ref' in json_data:
+                        version_data['source_control_ref'] = json_data['source_control_ref']
+
+                    version_index = _find_index(repo_objects_data[index].get('versions_data',[]), 'version', json_data['version'])
+                    if version_index != -1:
+                        # update the existing version
+                        if version_data:
+                            repo_objects_data[index]['versions_data'][version_index] = version_data
+                    else:
+                        # add the new version
+                        if 'versions_data' not in repo_objects_data[index]:
+                            repo_objects_data[index]['versions_data'] = [version_data] 
+                        else:
+                            repo_objects_data[index]['versions_data'].append(version_data) 
+                else:
+                    repo_objects_data.append(json_data)
+
+    # Remove duplicates from list 
+    if delete_objects:      
+        removal_list = delete_objects.split() if isinstance(delete_objects, str) else delete_objects
+        # find name field of object you want to delete
+        object_key = str(f'{object_typename}_name')
+        for removal_object in removal_list:
+            # Remove the JSON object(s) with the specified object_name value
+            repo_objects_data = [object_typename for object_typename in repo_objects_data if object_typename.get(object_key) != removal_object]
+
+    
+    # Replace 
+    if replace_objects:
+        paths = replace_objects.split() if isinstance(replace_objects, str) else replace_objects
+        for object_path in paths:
+            # clear suggect object field
+            repo_objects_data = []
```

Comment:
instead of copying the code from above I recommend you either make a helper function and call it, or modify the logic for `add_objects` as suggested above